neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/BaseNeo4jDialect.java

```diff
@@ -297,4 +300,18 @@ protected EntityKeyMetadata entityKeyMetadata(EntityKeyMetadata keyMetadata, Tup
 	}
 
 	protected abstract A createNeo4jAssociationQueries(EntityKeyMetadata ownerEntityKeyMetadata, AssociationKeyMetadata associationKeyMetadata);
+
+	protected Map.Entry<String, Map> buildProcedureQueryWithParams(String storedProcedureName,
+			ProcedureQueryParameters queryParameters) {
+		StringBuilder queryBuilder = new StringBuilder();
+		queryBuilder.append( "CALL " ).append( storedProcedureName ).append( "(" );
+		Map namedParams = new HashMap<>();
+		List parameters = queryParameters.getPositionalParameters();
+		for ( int i = 0; i < parameters.size(); i++ ) {
+			queryBuilder.append( "{a" + i + "}," );
```

Comment:
I think it's fine, I've just a curiosity: why add an 'a' in front of parameter name?

Reply:
Just not to cast value of parameter identifier to string manually on next line

Reply:
> Just not to cast value of parameter identifier to string manually on next line

Please, use the casting. I'm not a big fun of this tricks. They just make the code weird.
You will also be able to reuse (most of) the code for named parameters if you avoid it.